### ISSUE TITLE
Config::sanityCheck() bug

### ISSUE THREAD

**Issue by scoopxyz** [AcademySoftwareFoundation/OpenColorIO#536]
Small bug I encountered today, _sanityCheck_ doesn't catch the following case:

```
- !<Look>
  name: primary_dn
  process_space: log
  transform: !<FileTransform> {src: my_file.cdl, interpolation: linear}
  transform: !<FileTransform> {src: my_file.cdl, interpolation: linear}
```

Where the YAML "transform" tag is defined twice, behaviour was that it didn't error it just grabbed the last defined "transform" and used that FileTransform.

**Comment by YashAgrawal0** [comment]
@scoopxyz  I'd like to work on this issue. 

Can you help me how do I get stated ?

**Comment by doug-walker** [comment]
@YashAgrawal0, let's ask @hodoulp if this is actually a good first issue.  To me, I don't think it is.

First of all, SanityCheck would not be able to detect this problem since once the Look object has been parsed, there is no record of the second transform.  Furthermore, this is not an isolated problem, it basically applies to anything in the config file.  OCIOYaml.cpp generally doesn't look for duplicate keys, duplicates just overwrite what has already been  set.  So if we actually want to fix this it probably requires fixing the entire Yaml parsing system.  I'm sure it's doable, but not a good first issue IMHO.

**Comment by hodoulp** [comment]
@YashAgrawal0, @doug-walker you raised a good point about the long-term goal. But I think we can still move step-by-step. The code already implements some validation on missing 'mandatory' keywords so it could be used to validate that the keyword is only present once.

Concerning the `transform` and `inverse_transform` & `process_space` of the `Look` I think that it would be a valuable add to the OCIOYaml parser.

@YashAgrawal0, You already demonstrated your ability to work in the OCIO project, I would suggest you to now start thinking/writing your project submission. Contact any of the mentors if needed.

**Comment by hodoulp** [comment]
@rashil2000 The method `CheckDuplicates()` is now available to check for other duplicates.

**Comment by rashil2000** [comment]
@hodoulp Okay, what all other functions need this check? I'll put up a `CheckDuplicates()` call in each of them.

**Comment by hodoulp** [comment]
@rashil2000 Thanks for the help. But your focus has to be your summer project because you only have two weeks to write it, and you will certainly need some feedback during the process.

**Comment by rashil2000** [comment]
@hodoulp Can I work on this now?

**Comment by hodoulp** [comment]
@rashil2000 yes you can continue.

**Comment by rashil2000** [comment]
@hodoulp Can you tell me what other functions need the check? I'll put up a call in each.

**Comment by hodoulp** [comment]
@rashil2000 As you now have a project for GSoC & demonstrated your ability to create a valid pull request, I will change the question. Take some time and propose us something. But if you have question(s) or blocker(s), don't waste too much time and contact us.

Once you have something in mind, validate your proposal before going too far in the code.

**Comment by rashil2000** [comment]
@hodoulp Okay, I'll try doing some investigation and send a PR.

**Comment by michdolan** [comment]
Fixed in #993
